### DISCUSSION TITLE
How to enable esbuild output to be mjs?

### DISCUSSION THREAD

**Discussion by MattBidewell** [aws/aws-sam-cli#4616] [Questions] [answered]
I've seen in the source code this might be possible but struggling to figure out the exact key to use in `buildProperties`.

[example of source code of aws lambda builder](https://github.com/aws/aws-lambda-builders/blob/f0bc378c62184cdd2567983df52cce7dbdb9a62f/aws_lambda_builders/workflows/nodejs_npm_esbuild/esbuild.py#:~:text=%22-,out_extension,-%22%2C)

[Heres an example project where I want to use it](https://github.com/MattBidewell/sam-cli-esbuild-top-level-await)

EDIT: 
hold up, I think I've found a feature request for it! My bad

https://github.com/aws/aws-lambda-builders/issues/426

Hopefully its on its way! 👀 🥳

**Comment by mildaniel** [comment, marked as answer]
Hey @MattBidewell, thanks for bringing this up. This feature is on its way out, it was released in the [latest version of Lambda Builders](https://github.com/aws/aws-lambda-builders/releases/tag/v1.25.0) and will make its way to into a coming release of SAM-CLI. When it's released, you will need to simply add a new property to your template metadata.

```
BuildMethod: esbuild
BuildProperties:
    OutExtension: 
     - .js=.mjs
```

**Reply by MattBidewell** [reply]
Awesome! And once again thanks for the prompt reply.

Exciting features coming to the Sam CLI! Great work

**Comment by mildaniel** [comment]
This feature has now been released in version 1.72.0!
